Add GraphQL mention to 2.5: "Document API errors..." (#364)
i0natan/nodebestpractices: sections/errorhandling/documentingusingswagger.md

# Document RESTful API errors using Swagger

### One Paragraph Explainer

REST APIs return results using HTTP status codes, it’s absolutely required for the API user to be aware not only about the API schema but also about potential errors – the caller may then catch an error and tactfully handle it. For example, your API documentation might state in advance that HTTP status 409 is returned when the customer name already exists (assuming the API register new users) so the caller can correspondingly render the best UX for the given situation. Swagger is a standard that defines the schema of API documentation offering an eco-system of tools that allow creating documentation easily online, see print screens below

> If you have already adopted GraphQL for your API endpoints, chances are that your schema already contains guarantees as to how errors should look and be handled by your client-side tooling, but you can add comment-based documentation as well.

Reviewer: nit: Minor rewording tweak for clarity and an easier read.

```suggestion
If you have already adopted GraphQL for your API endpoints, chances are that your schema already contains guarantees as to what errors should look like and how they should be handled by your client-side tooling. In addition, you can also supplement them with comment-based documentation.
```

> If you have already adopted GraphQL for your API endpoints, chances are that your schema already contains guarantees as to what errors should look like and how they should be handled by your client-side tooling. In addition, you can also supplement them with comment-based documentation.

Reviewer 2: @BrunoScheufler To my limited understanding of Graph, it contains a strict definition of how & where Error should appear within the response (link below), shouldn't we:
1. Say this explicitly (not "chances are" rather "the Graph spec clearly...")
2. Show an example of Graph error
3. Consider recommending to supplement each error with _extensions.code_ to ensure it has a strong identifier

https://facebook.github.io/graphql/June2018/#sec-Errors

> GraphQL services may provide an additional entry to errors with key extensions. This entry, if set, must have a map as its value. This entry is reserved for implementors to add additional information to errors however they see fit, and there are no additional restrictions on its contents.

Author: I'll rephrase that section and include an example response 👍
